release-calendar/src/client/components/Calendar.tsx

```diff
@@ -23,23 +23,41 @@ import timeGridPlugin from '@fullcalendar/timegrid';
 
 export interface CalendarProps {
   events: EventSourceInput[];
+  selectedChannel: number;
 }
 
 export class Calendar extends React.Component<CalendarProps, {}> {
   render(): JSX.Element {
-    return (
-      <div className='calendar'>
-        <FullCalendar
-          defaultView='dayGridMonth'
-          header={{
-            left: 'prev,next',
-            center: 'title',
-            right: 'dayGridMonth,timeGridWeek,listWeek',
-          }}
-          plugins={[dayGridPlugin, timeGridPlugin]}
-          eventSources={this.props.events}
-        />
-      </div>
-    );
+    if (this.props.selectedChannel != null) {
```

Comment:
This feels like unnecessary duplication. Any way `events` can be set before rendering?

Reply:
yes it can be, will do!